src/aws_metadata.app.src

```diff
@@ -5,7 +5,8 @@
   {mod, {aws_metadata_app, []}},
   {applications,
    [kernel,
-    stdlib
+    stdlib,
+    aws
```

Comment:
When listed under `applications` an app is included as dependency in releases (when using rebar3/relx). Alternatively, we could add it to `included_applications` which has the same effect but doesn't start the application. For example, `aws-erlang` lists `jsx` under `applications`, `jsx` is also stateless. There's no harm in including it, it's just more explicit.